### DISCUSSION TITLE
Retrieve list of enums

### DISCUSSION THREAD

**Discussion by andyjamesn** [supabase/supabase#8818] [Questions]
Is there a way to retrieve a list of enums to use on the front end?

Say I have an enum called component_type for which enums is enough, I do not need a separate table for the types as there are only a few.

On the front end I have a select that allows the user to select a component type. It would be ideal if these values came from the database so I am wondering if there is a way to retrieve them?

**Comment by GaryAustin1** [comment]
You could write an rpc function and return an array using enum_range().
https://www.postgresql.org/docs/current/functions-enum.html
https://stackoverflow.com/questions/1616123/sql-query-to-get-all-values-a-enum-can-have
